File: app/src/main/java/me/pwcong/jpstart/mvp/presenter/MemoryFragmentPresenterImpl.java
package me.pwcong.jpstart.mvp.presenter;

import me.pwcong.jpstart.R;
import me.pwcong.jpstart.conf.Constants;
import me.pwcong.jpstart.mvp.model.BaseModel;
import me.pwcong.jpstart.mvp.model.MemoryFragmentModelImpl;
import me.pwcong.jpstart.mvp.view.BaseView;

/**
 * Created by Pwcong on 2016/10/5.
 */

public class MemoryFragmentPresenterImpl extends BasePresenter<BaseView.MemoryFragmentView> implements BasePresenter.MemoryFragmentPresenter {

    private BaseModel.MemoryFragmentModel model;

    public MemoryFragmentPresenterImpl(BaseView.MemoryFragmentView view) {
        super(view);
        model = new MemoryFragmentModelImpl();
    }

    @Override
    public void initMemoryFragment() {
        setDate(Constants.CATEGORY_QINGYIN);
    }

    @Override
    public void loadMore(int categiry) {

        setDate(categiry);
        view.showMsg(R.string.one_more_time);
    }

    @Override
    public void setDate(int category) {

        switch (category) {

            case Constants.CATEGORY_QINGYIN:

                view.setData(model.getQingYinWithoutHeader());

                break;
            case Constants.CATEGORY_ZHUOYIN:

                view.setData(model.getZhuoYinWithoutHeader());

                break;
            case Constants.CATEGORY_AOYIN:

                view.setData(model.getAoYinWithoutHeader());

                break;
            default:
                break;


        }

    }


}

File: app/src/main/java/com/github/pwcong/jpstart/mvp/presenter/impl/MemoryFragmentPresenterImpl.kt
package com.github.pwcong.jpstart.mvp.presenter.impl


import com.github.pwcong.jpstart.R
import com.github.pwcong.jpstart.constants.Constants
import com.github.pwcong.jpstart.mvp.model.BaseModel
import com.github.pwcong.jpstart.mvp.model.impl.MemoryFragmentModelImpl
import com.github.pwcong.jpstart.mvp.presenter.BasePresenter
import com.github.pwcong.jpstart.mvp.presenter.BasePresenter.MemoryFragmentPresenter
import com.github.pwcong.jpstart.mvp.view.BaseView.MemoryFragmentView

class MemoryFragmentPresenterImpl(view: MemoryFragmentView) :
    BasePresenter<MemoryFragmentView>(view),
    MemoryFragmentPresenter {
    private val model: BaseModel.MemoryFragmentModel = MemoryFragmentModelImpl()

    override fun initMemoryFragment() {
        setDate(Constants.CATEGORY_QINGYIN)
    }

    override fun loadMore(category: Int) {
        setDate(category)
        view.showMsg(R.string.one_more_time)
    }

    override fun setDate(category: Int) {
        when (category) {
            Constants.CATEGORY_QINGYIN -> view.setData(model.qingYinWithoutHeader)

            Constants.CATEGORY_ZHUOYIN -> view.setData(model.zhuoYinWithoutHeader)

            Constants.CATEGORY_AOYIN -> view.setData(model.aoYinWithoutHeader)

            else -> {}
        }
    }
}
